ecosystem/sep-0010.md

```diff
@@ -120,19 +125,28 @@ To validate the challenge transaction the following steps are performed by the s
 * decode the received input as a base64-urlencoded XDR representation of Stellar transaction envelope;
 * verify that transaction source account is equal to the server's signing key;
 * verify that transaction has time bounds set, and that current time is between the minimum and maximum bounds;
-* verify that transaction contains a single [Manage Data](https://www.stellar.org/developers/guides/concepts/list-of-operations.html#manage-data) operation and it's source account is not null;
-* verify that transaction envelope has a correct signature by server's signing key;
-* verify that transaction envelope has a correct signature by the operation's source account;
+* verify that transaction contains a single [Manage Data](https://www.stellar.org/developers/guides/concepts/list-of-operations.html#manage-data) operation and its source account is not null;
 * verify that transaction sequenceNumber is equal to zero;
+* verify that transaction envelope has a correct signature by server's signing key;
+* verify that client signatures on the transactions are signers of the operations source account;
+* verify that client signatures are correct;
+* verify that client signatures provide weight that meets the required threshold(s);
 * use operations's source account to determine the authenticating client and perform any additional service-specific validations.
 
+When determining the weight of the signers the server's signer should be excluded, even if it is a signer on the account. In most cases the server should verify that the signers meet all thresholds of the account by checking that the sum of the weights is equal or greater to the low, mid and high thresholds, and fail to issue a JWT if all thresholds are not met. Some servers may have use cases to issue JWTs for less than all thresholds.
```

Comment:
Maybe make it clear that the server is verifying that the client signature(s) meet all thresholds. Also, do you need to check all thresholds? That is, is it possible for for high >= medium >= low to be not true?

Also, maybe change "fail to issue a JWT if all thresholds are not met" to "should only issue a JWT if all thresholds are met" (minor nit, avoiding negatives)

Reply:
>do you need to check all thresholds? That is, is it possible for for high >= medium >= low to be not true?

It's definitely possible to configure a Stellar account with low, medium and high not in the order you described. I created an example here that does it and it works: https://repl.it/repls/UsableBurlyDesigns.

I've improved the language as you suggested in f00d22a.